lib/protobuf-c.cmake

```diff
@@ -0,0 +1,61 @@
+# Compile protobuf-c
+include(CheckLibraryExists)
+include(ExternalProject)
+
+if(BUILD_ONLY_DOCS)
+  # skip
+elseif(NOT BUILD_PROTOBUFC_LIB)
+  find_package(protobuf-c MODULE REQUIRED)
+  message("Found protobuf-c library: ${protobuf-c_LIBRARIES}")
+else()
+  # Download and Compile protobuf-c library at compile time using ExternalProject
+  # (e.g. when running `cmake --build` or `make`)
+  set(LIBPROTOBUFC_INSTALL_ROOT "${CMAKE_CURRENT_BINARY_DIR}/lib")
+  set(LIBPROTOBUFC_INSTALL_DIR "${LIBPROTOBUFC_INSTALL_ROOT}/protobuf-c")
+
+  if(BUILD_SHARED_LIBS)
+    set(configure_args "--enable-shared" "--disable-static" "--disable-protoc")
+  else()
+    set(configure_args "--enable-static" "--disable-shared" "--disable-protoc")
+  endif()
+
+  message("Downloading and compiling our own libprotobuf-c library")
+  ExternalProject_Add(
+    libprotobuf-c
+    URL https://github.com/protobuf-c/protobuf-c/releases/download/v1.4.1/protobuf-c-1.4.1.tar.gz
```

Comment:
Also, we should add a `URL_HASH SHA256=0123456789abcdef000000000000000000000000000000000000000000000000`, so that CMake skips trying to download the file again.

(it also means that the `protobuf-c` team can't just edit their `v1.4.1` tag in the future to switch to a malicious version)

Reply:
For now let's stay at the version 1.4.0 and compile it as the library is quite buggy and version 1.3.3 can introduce more bugs.